src/OrchardCore.Modules/OrchardCore.Security/Services/SecurityHeadersMiddleware.cs

```diff
@@ -13,6 +13,11 @@ public SecurityHeadersMiddleware(SecurityHeadersOptions options, RequestDelegate
         {
             _options = options;
             _next = next;
+
+            foreach (var provider in _options.HeaderPolicyProviders)
+            {
+                provider.InitializePolicy();
+            }
```

Comment:
I don't see why this is better than doing the same thing in each provider's ctor and not adding an init method.

Reply:
They are more like builders than providers, and in the current pattern they are build in the ctor of `SecurityHeadersOptions`, and at this time the other policy fields are not yet populated.

Reply:
I see it now.